middleware.go

```diff
@@ -73,6 +89,60 @@ func EchoFirstTraceNodeInfo(propagator propagation.TextMapPropagator) func(http.
 	}
 }
 
+// findWRPTraceHeaders looks for trace headers in WRP messages
+// Returns a context propagated with the trace information
+func findWRPTraceHeaders(ctx context.Context, r *http.Request) context.Context {
+
+	// Check for trace info in Xmidt headers
+	var traceHeaders []string
+	xmidtHeaders := r.Header.Values("X-midt-Headers")
+	if len(xmidtHeaders) != 0 {
+		traceHeaders = xmidtHeaders
+	} else {
+
+		// Check for trace info in the request body
+		reader := r.Body
+		bodyData, err := io.ReadAll(reader)
+		if err != nil {
+			log.Printf("failed to read body: %s", err)
+		}
+
+		r.Body, r.GetBody = xhttp.NewRewindBytes(bodyData)
```

Comment:
This is the only function that is needed from webpa-common ... but it's a bit tangled.  The `r.Body` is a `io.ReadCloser` so I'm not sure if the `bufio.Reader` is going to work here.  @johnabass any suggestions?

Reply:
We should defiinitely remove the rewind stuff here.  It's really just an optimization hack to allow reuse of the same buffer when redirects happen.  I would first try to make `bufio.Reader` work, in tandem with `io.NopCloser`.  Failing that, we should move this to `httpaux`.